File: lib/data/models/delivery_route.dart
class DeliveryRoute {
  const DeliveryRoute({
    required this.id,
    required this.name,
    this.active = true,
  });

  final String id;
  final String name;
  final bool active;

  DeliveryRoute copyWith({
    String? name,
    bool? active,
  }) {
    return DeliveryRoute(
      id: id,
      name: name ?? this.name,
      active: active ?? this.active,
    );
  }
}

File: backend/src/models/DeliveryRoute.js
const mongoose = require('mongoose');

const deliveryRouteSchema = new mongoose.Schema(
  {
    routeId: { type: String, required: true, unique: true },
    shopId: { type: String, required: true },
    name: { type: String, required: true, trim: true },
    active: { type: Boolean, default: true },
    sortOrder: { type: Number, default: 0 },
  },
  { timestamps: true },
);

deliveryRouteSchema.index({ shopId: 1, name: 1 }, { unique: true });
deliveryRouteSchema.index({ shopId: 1, active: 1 });

module.exports = mongoose.model('DeliveryRoute', deliveryRouteSchema);
